service/dap/server.go

```diff
@@ -1376,23 +1371,32 @@ func (s *Server) onEvaluateRequest(request *dap.EvaluateRequest) {
 			// As a shortcut also express the return values as a single string.
 			retVarsAsStr := ""
 			for _, v := range retVars {
-				retVarsAsStr += s.convertVariableToString(v, loadConfig) + ", "
+				retVarsAsStr += s.convertVariableToString(v) + ", "
 			}
 			response.Body = dap.EvaluateResponseBody{
 				Result:             strings.TrimRight(retVarsAsStr, ", "),
 				VariablesReference: s.variableHandles.create(&fullyQualifiedVariable{retVarsAsVar, "", false /*not a scope*/}),
 			}
 		}
 	} else { // {expression}
-		exprVar, err := s.debugger.EvalVariableInScope(goid, frame, 0, request.Arguments.Expression, loadConfig)
+		exprVar, err := s.debugger.EvalVariableInScope(goid, frame, 0, request.Arguments.Expression, DefaultLoadConfig)
 		if err != nil {
 			s.sendErrorResponseWithOpts(request.Request, UnableToEvaluateExpression, "Unable to evaluate expression", err.Error(), showErrorToUser)
 			return
 		}
+		// If we are in interactive mode (probably with sufficient space to display long strings),
+		// load the string with a more generous load config.
+		if ctxt := request.Arguments.Context; (ctxt == "repl" || ctxt == "variables") && exprVar.Kind == reflect.String {
```

Comment:
Do you mean to do this for scalar string vars only? Weren't users who complained about this feature storing their strings in some kind of data structures? For example, https://github.com/golang/vscode-go/issues/1318. This might not be unintuitive. We would still be limited by recursion limit and array limits (if you don't change them). Have you actually tried testing this to see if it would be too much?

Reply:
I have been thinking more about this. I guess with proper documentation, we could advertise loading specific string fields with "Copy Value" and "Copy as Expression". And this would certainly make it safer performance-wise.

Reply:
This is focusing only on string value retrieved with evaluate + copy value context only.